### PR TITLE
Fix primary positioning

### PR DESCRIPTION
**Pull request by ItzNotABug**
## What does this PR do?

1. Found an edge case where the primary column `$id` moved to last, this fixes it.
2. Optimized the ID calc logic. Previously showed `N` requestFrameAnimation messages in console, batched it now.

## Test Plan

Manual.

## Related PRs and Issues

N/A.

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes.

## Summary by CodeRabbit

* **Bug Fixes**
  * Text truncation is now more accurate and stable, preventing over/under-truncation.
  * Truncated text updates on window resize without visual jank thanks to debounced handling and queued checks.
  * Moving a column now reliably promotes the moved item to the first position when applicable.

* **Refactor**
  * Truncation and resize logic optimized for performance and smoother UI updates.

### CODE REVIEW

File: src/lib/components/id.svelte [hunk 1]
```diff
@@ -1,41 +1,73 @@
...
+    let batchPromise: Promise<void> | null = null;
+
+    async function processBatch() {
+        await tick();
+        batchQueue.forEach((fn) => fn());
+
+        // clear queue!
+        batchQueue.clear();
+        batchPromise = null;
+    }
```
[the remark this answers is not in the document]
Comment (ItzNotABug) [reply src/lib/components/id.svelte]: makes sense but why grab everything in a try/finally. the `fn()` is anyway inside a try/catch, right? whats the point unless I am missing something.

**Review by TorstenDittmann — APPROVED** [review]
